bundles/org.openhab.binding.semsportal/src/main/java/org/openhab/binding/semsportal/internal/SEMSPortalHandler.java

```diff
@@ -0,0 +1,268 @@
+                stationOnline = currentStatus.isOnline();
+                triggerChannel(channelUID, OnOffType.from(stationOnline).name());
+            }
+        } else if (CHANNEL_TODAY_INCOME.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getDayIncome(currentStatus));
+        } else if (CHANNEL_TOTAL_INCOME.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getTotalIncome(currentStatus));
+        } else if (CHANNEL_LASTUPDATE.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getLastUpdate(currentStatus));
+        }
+    }
+
+    private void ensureRecentStatus() {
+        if (lastUpdate.isBefore(LocalDateTime.now().minusMinutes(MAX_STATUS_AGE_MINUTES))) {
+            updateStation();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SEMSPortalConfiguration.class);
+        if (!config.isProperlyInitialized()) {
+            logger.warn("Username, password or station missing. Please provide the right values in the configuration.");
+            updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Check username / password / station settings");
+            return;
+        }
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            try {
+                login();
+                if (loggedIn) {
+                    pollingJob = scheduler.scheduleWithFixedDelay(this::updateStation, 0, config.update,
+                            TimeUnit.MINUTES);
+                }
+            } catch (Exception e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                        "Error when loggin in. Check your username and password");
+            }
+        });
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+        }
+        super.dispose();
+    }
+
+    private void login() {
+        loggedIn = false;
+        String payload = gson.toJson(new LoginRequestBody(config.username, config.password));
+        String response = sendPost(LOGIN_URL, payload);
+        if (response == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Invalid response from SEMS portal");
+            return;
+        }
+        SEMSLoginResponse loginResponse = gson.fromJson(response, SEMSLoginResponse.class);
+        if (loginResponse == null) {
+            logger.debug("SEMSPortal answer not understood: {}", loginResponse);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Check username / password");
+            return;
+        }
+        if (loginResponse.isOk()) {
+            logger.info("Successfuly logged in to SEMS portal");
```

Comment:
Perhaps make this DEBUG level, or remove it as can it be assumed that this has occurred if the thing came ONLINE?